SignalServiceKit/src/Network/API/Requests/TSRegisterForPushRequest.m

```diff
@@ -1,29 +1,30 @@
 //
-//  TSRegisterForPushRequest.m
-//  TextSecureiOS
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
 //
-//  Created by Frederic Jacobs on 10/13/13.
-//  Copyright (c) 2013 Open Whisper Systems. All rights reserved.
-//
-
-#import "TSConstants.h"
 
 #import "TSRegisterForPushRequest.h"
+#import "TSConstants.h"
 
 @implementation TSRegisterForPushRequest
 
 - (id)initWithPushIdentifier:(NSString *)identifier voipIdentifier:(NSString *)voipId {
     self =
         [super initWithURL:[NSURL URLWithString:[NSString stringWithFormat:@"%@/%@", textSecureAccountsAPI, @"apn"]]];
 
+    if (!self) {
+        return nil;
+    }
+
     self.HTTPMethod = @"PUT";
 
-    self.parameters = [@{ @"apnRegistrationId" : identifier } mutableCopy];
+    NSMutableDictionary *parameters = [@{ @"apnRegistrationId" : identifier } mutableCopy];
 
     if (voipId) {
-        [self.parameters setObject:voipId forKeyedSubscript:@"voipRegistrationId"];
```

Comment:
This is some ancient code. I don't think we ever expect this to happen any more. Even on devices that are known not to support voip pushes, like simulators, we return a fake token.

Can we make the voipIdentifier arg non-null and verify nothing else breaks?

We can put this in a card for later if you want.

Reply:
Done.